scrapy/spiders/sitemap.py

```diff
@@ -49,6 +49,13 @@ def _parse_sitemap(self, response):
                 logger.warning("Ignoring invalid sitemap: %(response)s",
                                {'response': response}, extra={'spider': self})
                 return
+            if response.url.endswith('.txt'):
+                for loc in iterloc_text_sitemap(response.text):
+                    for r, c in self._cbs:
```

Comment:
:lipstick: This part might, though, as it’s repeated below.

```
for r, c in self._cbs:
    if r.search(loc):
        yield Request(loc, callback=c)
        break
```

Could be extracted as:

```
    def _url_request(self, url):
        for regex, callback in self._cbs:
            if regex.search(url):
                return Request(url, callback=callback)

    # …

                    request = self._url_request(loc)
                    if request:
                        yield request
```